Android: modernize code

PatrykMis opened the pull request:
* Restructure Gradle / Android Studio project
* Convert build logic to fully Kotlin DSL
* code restructure (reformat, optimize imports and cleanup)
* avoid deprecated APIs and migrate to new ones (if needed)
* Code improvements (following system's light/dark theme, enabling TTS pitch / rate / volume controls, etc.)

PatrykMis committed:
Android: restructure project

Signed-off-by: Patryk Mis <foss@patrykmis.com>

PatrykMis committed:
Android: convert build scripts to Kotlin DSL

Signed-off-by: Patryk Mis <foss@patrykmis.com>

beqabeqa473 commented:
• code restructure (reformat, optimize imports and cleanup)
I 'm not sure if some bits of android port was touched

On 4/22/23, Patryk Miś ***@***.***> wrote:
> * Restructure Gradle / Android Studio project
> * Convert build logic to fully Kotlin
> * code restructure (reformat, optimize imports and cleanup)
> * avoid deprecated APIs and migrate to new ones (if needed)
>
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/espeak-ng/espeak-ng/pull/1709
>
> -- Commit Summary --
>
>   * Android: restructure project
>
> -- File Changes --
>
>     M .gitignore (3)
>     A android/app/build.gradle (90)
>     R
> android/app/src/androidTest/java/com/reecedunn/espeak/test/CheckVoiceDataTest.java
> (0)
>     R
> android/app/src/androidTest/java/com/reecedunn/espeak/test/SpeechSynthesisTest.java
> (0)
>     R
> android/app/src/androidTest/java/com/reecedunn/espeak/test/TextToSpeechServiceTest.java
> (0)
>     R
> android/app/src/androidTest/java/com/reecedunn/espeak/test/TextToSpeechTest.java
> (0)
>     R
> android/app/src/androidTest/java/com/reecedunn/espeak/test/TextToSpeechTestCase.java
> (0)
>     R
> android/app/src/androidTest/java/com/reecedunn/espeak/test/TtsMatcher.java
> (0)
>     R
> android/app/src/androidTest/java/com/reecedunn/espeak/test/VoiceData.java
> (0)
>     R
> android/app/src/androidTest/java/com/reecedunn/espeak/test/VoiceSettingsTest.java
> (0)
>     R
> android/app/src/androidTest/java/com/reecedunn/espeak/test/VoiceVariantTest.java
> (0)
>     R android/app/src/main/AndroidManifest.xml (4)
>     R android/app/src/main/cpp/CMakeLists.txt (7)
>     R android/app/src/main/cpp/include/Log.h (0)
>     R android/app/src/main/cpp/include/config.h (0)
>     R android/app/src/main/cpp/jni/eSpeakService.c (0)
>     R android/app/src/main/java/com/reecedunn/espeak/CheckVoiceData.java
> (0)
>     R android/app/src/main/java/com/reecedunn/espeak/DownloadVoiceData.java
> (0)
>     R android/app/src/main/java/com/reecedunn/espeak/EspeakApp.java (0)
>     R android/app/src/main/java/com/reecedunn/espeak/FileListAdapter.java
> (0)
>     R android/app/src/main/java/com/reecedunn/espeak/FileUtils.java (0)
>     R android/app/src/main/java/com/reecedunn/espeak/GetSampleText.java (0)
>     R
> android/app/src/main/java/com/reecedunn/espeak/InformationListAdapter.java
> (0)
>     R
> android/app/src/main/java/com/reecedunn/espeak/ResourceIdListAdapter.java
> (0)
>     R android/app/src/main/java/com/reecedunn/espeak/SpeechSynthesis.java
> (0)
>     R android/app/src/main/java/com/reecedunn/espeak/TtsService.java (0)
>     R
> android/app/src/main/java/com/reecedunn/espeak/TtsSettingsActivity.java (0)
>     R android/app/src/main/java/com/reecedunn/espeak/Voice.java (0)
>     R android/app/src/main/java/com/reecedunn/espeak/VoiceSettings.java (0)
>     R android/app/src/main/java/com/reecedunn/espeak/VoiceVariant.java (0)
>     R android/app/src/main/java/com/reecedunn/espeak/eSpeakActivity.java
> (0)
>     R
> android/app/src/main/java/com/reecedunn/espeak/preference/ImportVoicePreference.java
> (0)
>     R
> android/app/src/main/java/com/reecedunn/espeak/preference/SeekBarPreference.java
> (0)
>     R
> android/app/src/main/java/com/reecedunn/espeak/preference/SpeakPunctuationPreference.java
> (0)
>     R
> android/app/src/main/java/com/reecedunn/espeak/preference/VoiceVariantPreference.java
> (0)
>     R android/app/src/main/res/drawable-hdpi/icon.png (0)
>     R android/app/src/main/res/drawable-ldpi/icon.png (0)
>     R android/app/src/main/res/drawable-mdpi/icon.png (0)
>     R android/app/src/main/res/drawable-xhdpi/icon.png (0)
>     R android/app/src/main/res/layout/download_voice_data.xml (0)
>     R android/app/src/main/res/layout/import_voice_preference.xml (0)
>     R android/app/src/main/res/layout/information_view.xml (0)
>     R android/app/src/main/res/layout/main.xml (0)
>     R android/app/src/main/res/layout/seekbar_preference.xml (0)
>     R android/app/src/main/res/layout/speak_punctuation_preference.xml (0)
>     R android/app/src/main/res/layout/voice_variant_preference.xml (0)
>     R android/app/src/main/res/menu/options.xml (0)
>     R android/app/src/main/res/values-af/strings.xml (0)
>     R android/app/src/main/res/values-am/strings.xml (0)
>     R android/app/src/main/res/values-ar/strings.xml (0)
>     R android/app/src/main/res/values-bg/strings.xml (0)
>     R android/app/src/main/res/values-ca/strings.xml (0)
>     R android/app/src/main/res/values-cs/strings.xml (0)
>     R android/app/src/main/res/values-da/strings.xml (0)
>     R android/app/src/main/res/values-de/strings.xml (0)
>     R android/app/src/main/res/values-el/strings.xml (0)
>     R android/app/src/main/res/values-en-rGB/strings.xml (0)
>     R android/app/src/main/res/values-es-rUS/strings.xml (0)
>     R android/app/src/main/res/values-es/strings.xml (0)
>     R android/app/src/main/res/values-fa/strings.xml (0)
>     R android/app/src/main/res/values-fi/strings.xml (0)
>     R android/app/src/main/res/values-fr/strings.xml (0)
>     R android/app/src/main/res/values-hi/strings.xml (0)
>     R android/app/src/main/res/values-hr/strings.xml (0)
>     R android/app/src/main/res/values-hu/strings.xml (0)
>     R android/app/src/main/res/values-in/strings.xml (0)
>     R android/app/src/main/res/values-it/strings.xml (0)
>     R android/app/src/main/res/values-iw/strings.xml (0)
>     R android/app/src/main/res/values-ja/strings.xml (0)
>     R android/app/src/main/res/values-ko/strings.xml (0)
>     R android/app/src/main/res/values-lt/strings.xml (0)
>     R android/app/src/main/res/values-lv/strings.xml (0)
>     R android/app/src/main/res/values-ms/strings.xml (0)
>     R android/app/src/main/res/values-nb/strings.xml (0)
>     R android/app/src/main/res/values-nl/strings.xml (0)
>     R android/app/src/main/res/values-pl/strings.xml (0)
>     R android/app/src/main/res/values-pt-rBR/strings.xml (0)
>     R android/app/src/main/res/values-pt/strings.xml (0)
>     R android/app/src/main/res/values-ro/strings.xml (0)
>     R android/app/src/main/res/values-ru/strings.xml (0)
>     R android/app/src/main/res/values-sk/strings.xml (0)
>     R android/app/src/main/res/values-sl/strings.xml (0)
>     R android/app/src/main/res/values-sr/strings.xml (0)
>     R android/app/src/main/res/values-sv/strings.xml (0)
>     R android/app/src/main/res/values-sw/strings.xml (0)
>     R android/app/src/main/res/values-th/strings.xml (0)
>     R android/app/src/main/res/values-tl/strings.xml (0)
>     R android/app/src/main/res/values-tr/strings.xml (0)
>     R android/app/src/main/res/values-uk/strings.xml (0)
>     R android/app/src/main/res/values-v21/styles.xml (0)
>     R android/app/src/main/res/values-vi/strings.xml (0)
>     R android/app/src/main/res/values-zh-rCN/strings.xml (0)
>     R android/app/src/main/res/values-zh-rTW/strings.xml (0)
>     R android/app/src/main/res/values-zu/strings.xml (0)
>     R android/app/src/main/res/values/donottranslate.xml (0)
>     R android/app/src/main/res/values/strings.xml (0)
>     R android/app/src/main/res/values/styles.xml (0)
>     R android/app/src/main/res/xml/preferences.xml (0)
>     R android/app/src/main/res/xml/tts_engine.xml (0)
>     M android/build.gradle (113)
>     D android/eSpeakTests/.classpath (9)
>     D android/eSpeakTests/.project (34)
>     D android/eSpeakTests/.settings/org.eclipse.jdt.core.prefs (4)
>     D android/eSpeakTests/AndroidManifest.xml (17)
>     D android/eSpeakTests/proguard-project.txt (20)
>     D android/eSpeakTests/project.properties (14)
>     D android/eSpeakTests/res/drawable-hdpi/ic_launcher.png (0)
>     D android/eSpeakTests/res/drawable-ldpi/ic_launcher.png (0)
>     D android/eSpeakTests/res/drawable-mdpi/ic_launcher.png (0)
>     D android/eSpeakTests/res/drawable-xhdpi/ic_launcher.png (0)
>     D android/eSpeakTests/res/values/strings.xml (6)
>     A android/gradle.properties (21)
>     M android/settings.gradle (15)
>
> -- Patch Links --
>
> https://github.com/espeak-ng/espeak-ng/pull/1709.patch
> https://github.com/espeak-ng/espeak-ng/pull/1709.diff
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/espeak-ng/espeak-ng/pull/1709
> You are receiving this because you are subscribed to this thread.
>
> Message ID: ***@***.***&gt;
>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

PatrykMis commented:
This is draft pull request (in progress). I've made it just to notice developers about work in progress.

alex19EP commented:
not sure if I like all those file mooving.

PatrykMis commented:
Why?

PatrykMis committed:
Android: small build tasks improvement

Signed-off-by: Patryk Mis <foss@patrykmis.com>

alex19EP commented:
because at the moment it's not of any use, we don't use different gradle modules...

PatrykMis committed:
Android: code reformat and cleanup

Signed-off-by: Patryk Mis <foss@patrykmis.com>

PatrykMis committed:
Resolve task dependency problems

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: properly configure deprecated instrumented tests

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: migrate resources to AppCompat

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: migrate sources to AppCompat

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: migrate to AndroidX

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis commented:
Hey @alex19EP , appreciate the feedback. Here's why I went with default Android Studio and Gradle structure:

1. It's Familiar: Most Android devs know this structure, making it easier to jump in.
2. Android Studio Benefits: The IDE has tools tailored for this setup, so why not make the most of them?
3. Simpler Builds: Less custom stuff in Gradle means fewer headaches and smoother builds.
4. Better Maintenance: Sticking closer to the default makes future updates smoother.
5. Clear Layout: Easier to find code, resources, and configs.
6. Resource Compatibility: Most Android docs and tutorials follow this structure. Easier for us to reference. I've updated scripts from outside of Android folder, too.
7. Community Help: Need a hand? More devs will get our setup.
8. Fewer Mistakes: Tried and tested structure = less room for slip-ups.

Just to highlight a few points. I think it sets us up well for the future, but I'm keen to hear your thoughts!

PatrykMis committed:
Android: migrate SeekBar

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: migrate to Version Catalogs, unify Java versions, bump deps

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: bump NDK to v26.1.10909125

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: bump compileSdk and buildTools versions

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: update .gitignore excluding keystore properties file

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: enable signing for release if keystore properties file is present

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: add `.dev` suffix to app id for debug builds

This makes possible for debug and release builds to coexist on the same device.

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: enable v1 signing explicitly

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: specify app name and version suffix for debug type

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: distinguish type from version

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: bump AGP to v8.2.0

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: build - rewrite checkData

Signed-off-by: Patryk Miś <foss@patrykmis.com>

alex19EP commented:
hello @PatrykMis looked at your pr again. some things I really like. however, I absolutely do not want to switch from groovy to kotlin. Don't get me wrong, if the entire project were in Kotlin, it would be logical, but now it looks like a change for the sake of change.

also I created and will merge #1863 so you should rebace your pr after it.

thanks.

PatrykMis committed:
Android: Update Gradle and AGP

Signed-off-by: Patryk Miś <foss@patrykmis.com>

PatrykMis committed:
Android: rewrite build tasks order

Signed-off-by: Patryk Miś <foss@patrykmis.com>

pvagner commented:
Dear @PatrykMis I just would like to thank you for this PR. It's amazing and it enables me to build the espeak android app on up to date arch linux install with android-sdk 34.
I am running it for a few days on my device and I am happy I can again build the app for my-self while testing my language updates.

PatrykMis commented:
Yes, I'll try to finalize this until half of october, sorry for the very long delay.

amirmahdifard commented:
I dislike kotlin dsl. I'd like this to continue as groovy and light, small and as simple as possible.
